### PR TITLE
feat: openssl investigation follow-up

### LINKED ISSUE #16962: OS Support: improve SSL detection on `debian` distros
When downloading the Prisma engines, we're currently resolving the `openssl` version with:
- `ls -l /lib64 | grep ssl`
- `ls -l /usr/lib64 | grep ssl`
- `openssl version -v`

However, Debian, Ubuntu, and similar distros define the `libssl.so.*` library in a specific path:

```console
"/usr/lib/$(uname -m)-linux-gnu/libssl.so.*"
```

We should prefer extracting the `openssl` version from that path.

### PR DESCRIPTION
**Pull request by jkomyno**
Draft investigation inspired by [this Slack thread](https://prisma-company.slack.com/archives/C04DG9JL6TV/p1672770455111859).

This adds:
- debug (`prisma:get-platform`) logging when resolving which `binaryTarget` Prisma should download
- uses `ls -r` and `ldconfig -p | ... | grep ssl | sort -r` to retrieve the newest available versions of `openssl` first
- RHEL-specific libssl paths to default to

This also closes https://github.com/prisma/prisma/issues/16962, as we now try both `/usr/lib/` and `/lib` baseline paths when looking for `libssl.so*`

### CODE REVIEW

File: packages/get-platform/src/getPlatform.ts [hunk 1]
```diff
@@ -151,60 +156,75 @@ type GetOpenSSLVersionParams = {
  * On Linux, returns the OpenSSL version excluding the patch version, e.g. "1.1.x".
  * Reading the version from the libssl.so file is more reliable than reading it from the openssl binary.
  * This function never throws.
+ *
+ * TODO: we should probably validate the output of this function, and ensure it's contained in `supportedLibSSLVersions`
```
Comment (Jolg42) [inline packages/get-platform/src/getPlatform.ts:160]: Is that a TODO for now or later?

File: packages/get-platform/src/getPlatform.ts [hunk 2]
```diff
@@ -151,60 +156,75 @@ type GetOpenSSLVersionParams = {
...
+    'ls -r /lib64 | grep ssl',
+    'ls -r /usr/lib64 | grep ssl',
   ])
 
-  if (libsslVersion) {
-    const matchedVersion = parseLibSSLVersion(libsslVersion)
-    if (matchedVersion) {
-      return matchedVersion
+  if (libsslFilename) {
+    debug(`Found libssl.so file using "ldconfig" and other generic paths: ${libsslFilenameFromSpecificPath}`)
```
Comment (Jolg42) [inline packages/get-platform/src/getPlatform.ts:206]:
```suggestion
    debug(`Found libssl.so file using "ldconfig" or other generic paths: ${libsslFilenameFromSpecificPath}`)
```
nit

File: packages/get-platform/src/getPlatform.ts (hunk 1 above)
Comment (jkomyno) [reply packages/get-platform/src/getPlatform.ts:160]: For https://github.com/prisma/prisma/issues/17154

**Review by Jolg42 — APPROVED** [review]
This looks good! And nice to have more debug logs for this!

**Pull request merged**